src/blocking/i2c.rs

```diff
@@ -127,3 +127,26 @@ pub trait WriteIterRead {
     where
         B: IntoIterator<Item = u8>;
 }
+
+/// Actions for transactional I2C trait
+///
+/// This allows composition of I2C operations into a single bus transaction
+#[cfg(feature = "unproven")]
+#[derive(Debug, PartialEq)]
+pub enum Actions<'a> {
+    /// Read data into the provided buffer
+    Read(&'a mut [u8]),
+    /// Write data from the provided buffer
+    Write(&'a [u8]),
+}
+
+/// Transactional trait allows multiple actions to be executed
+/// as part of a single I2C transaction
+#[cfg(feature = "unproven")]
+pub trait Transactional {
+    /// Associated error type
+    type Error;
+
+    /// Execute the provided actions against the provided I2C address
+    fn exec(&mut self, addr: u8, actions: &[Actions]) -> Result<(), Self::Error>;
```

Comment:
do we like the name `exec` here, or would something else be better?

Reply:
I think it is fine, except someone else has a better idea.